patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2525,6 +2525,26 @@
  		}
  
  		public void DrawItems() {
+@@ -28356,6 +_,19 @@
+ 				Rain rain = Main.rain[i];
+ 				if (rain.active) {
+ 					value.X = rain.type * 4;
++
++					if (rain.waterStyle >= maxLiquidTypes) {
++						value2 = LoaderManager.Get<WaterStylesLoader>().Get(rain.waterStyle).GetRainTexture().Value;
++						if (value2 == null) {
++							if (flag)
```

Comment:
Suggest reducing nesting by using short circuit 
(Value2 == null && flag)

Reply:
That would be functionally different, there's a `continue` in there as well which does not require `flag`.